File: go-client-sdk/pkg/models/operations/getintegrationsplatformkeybranding.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetIntegrationsPlatformKeyBrandingPathParams struct {
	PlatformKey string `pathParam:"style=simple,explode=false,name=platformKey"`
}

type GetIntegrationsPlatformKeyBrandingSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetIntegrationsPlatformKeyBrandingRequest struct {
	PathParams GetIntegrationsPlatformKeyBrandingPathParams
	Security   GetIntegrationsPlatformKeyBrandingSecurity
}

type GetIntegrationsPlatformKeyBrandingResponse struct {
	CodatPublicAPIModelsClientsIntegrationBrandingModel *shared.CodatPublicAPIModelsClientsIntegrationBrandingModel
	ContentType                                         string
	StatusCode                                          int
}

File: typescript-client-sdk/src/sdk/models/operations/getintegrationsplatformkeybranding.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetIntegrationsPlatformKeyBrandingPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=platformKey" })
  platformKey: string;
}

export class GetIntegrationsPlatformKeyBrandingSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetIntegrationsPlatformKeyBrandingRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetIntegrationsPlatformKeyBrandingPathParams;

  @SpeakeasyMetadata()
  security: GetIntegrationsPlatformKeyBrandingSecurity;
}

export class GetIntegrationsPlatformKeyBrandingResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsClientsIntegrationBrandingModel?: shared.CodatPublicApiModelsClientsIntegrationBrandingModel;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}